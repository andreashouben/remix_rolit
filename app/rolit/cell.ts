export type Color = "R" | "Y" | "B" | "G";
export type CellValue = Color | "O"

export class Cell {

  private color: CellValue = "O";

  value() {
    return this.color;
  }

  turnRed() {
    this.color = "R";
  }

  turnYellow() {
    this.color = "Y";
  }

  turnBlue() {
    this.color = "B";
  }

  turnGreen() {
    this.color = "G";
  }
}