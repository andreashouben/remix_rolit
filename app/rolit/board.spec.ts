import { Cell } from "~/rolit/cell";

class Board {
  getGrid() {
    const createEmptyRow = () => [new Cell(),new Cell(),new Cell(),new Cell(),
      new Cell(),new Cell(),new Cell(),new Cell()]
    const board = [
      createEmptyRow(),
      createEmptyRow(),
      createEmptyRow(),
      createEmptyRow(),
      createEmptyRow(),
      createEmptyRow(),
      createEmptyRow(),
      createEmptyRow(),
    ];
    board[3][3].turnRed();
    board[3][4].turnGreen();
    board[4][3].turnBlue();
    board[4][4].turnYellow();
    return board;
  }
}

const printGrid = (grid: Cell[][]): string =>
  grid
    .map(row => row
      .map(cell => cell.value())
      .join(" "))
    .join("\n");

describe("board", () => {
  it("initializes as 8x8 grid with the four colors in the middle", () => {
    const board = new Board();

    const grid = board.getGrid();

    expect(printGrid(grid)).toMatchSnapshot()
  });
});