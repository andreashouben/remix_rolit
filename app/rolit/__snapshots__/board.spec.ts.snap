// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`board > initializes as 8x8 grid with the four colors in the middle 1`] = `
"O O O O O O O O
O O O O O O O O
O O O O O O O O
O O O R G O O O
O O O B Y O O O
O O O O O O O O
O O O O O O O O
O O O O O O O O"
`;
